sql/src/main/java/io/crate/analyze/OrderBy.java

```diff
@@ -64,6 +65,20 @@ public void normalize(EvaluatingNormalizer normalizer) {
         normalizer.normalizeInplace(orderBySymbols);
     }
 
+
+    public OrderBy subset(Collection<Integer> positions){
+        List<Symbol> orderBySymbols = new ArrayList<>(positions.size());
+        Boolean[] nullsFirst = new Boolean[positions.size()];
+        boolean[] reverseFlags = new boolean[positions.size()];
+        int pos = 0;
+        for (Integer i : positions) {
+            orderBySymbols.add(this.orderBySymbols.get(i));
+            nullsFirst[pos] = this.nullsFirst[i];
+            reverseFlags[pos] = this.reverseFlags[i];
```

Comment:
pos is never incremented? can you add a test where subset is called with 2 positions?